AllTheThings.lua

```diff
@@ -16746,6 +16746,53 @@ RowOnEnter = function (self)
 		if reference.speciesID then
 			local progress, total = C_PetJournal.GetNumCollectedInfo(reference.speciesID);
 			if total then GameTooltip:AddLine(tostring(progress) .. " / " .. tostring(total) .. L["COLLECTED_STRING"]); end
+			
+			if progress > 0 and app.Settings:GetTooltipSetting("PetBattleQuality") then
+				local speciesName = C_PetJournal.GetPetInfoBySpeciesID(reference.speciesID);
+				
+				-- Set the journal search filter to the pet name and the category for example duskwood or in-game shop
+				-- This is done so we limit how many pets will show up.
+				-- If category text is formatted we only search for name since the formatting will break the search
+				if string.find(reference.parent.parent.text, "|") == nil then
```

Comment:
Not sure the use case for this `.parent.parent`... this will cause `nil` references in some situations (i.e. a tooltip of a popout window of a specific Battle Pet)
Why does the parent hierarchy relate to the search for the Pet?

Reply:
Fixed this checking if parent is not nil and parent.parent is not nil